koku/masu/database/trino_sql/reporting_ocpawscostlineitem_daily_summary.sql

```diff
@@ -198,6 +201,19 @@ SELECT cast(uuid() as varchar) as uuid,
     nullif(aws.product_region, '') as region,
     max(nullif(aws.pricing_unit, '')) as unit,
     sum(aws.lineitem_usageamount) as usage_amount,
+    -- Determine network direction
+    CASE
+        -- Is this a network record?
+        WHEN max(aws.lineitem_productcode) = 'AmazonEC2' AND max(aws.product_productfamily) = 'Data Transfer' THEN
+            -- Yes, it's a network record. What's the direction?
+            CASE
+                WHEN position('in-bytes' IN lower(max(aws.lineitem_usagetype))) > 0 THEN 'IN'
+                WHEN position('out-bytes' IN lower(max(aws.lineitem_usagetype))) > 0 THEN 'OUT'
+                WHEN (position('regional-bytes' IN lower(max(aws.lineitem_usagetype))) > 0 AND position('-in' IN lower(max(lineitem_operation))) > 0) THEN 'IN'
+                WHEN (position('regional-bytes' IN lower(max(aws.lineitem_usagetype))) > 0 AND position('-out' IN lower(max(lineitem_operation))) > 0) THEN 'OUT'
+                ELSE NULL
+            END
+    END AS data_transfer_direction,
```

Comment:
This SQL is grouped by these items below, are these enough to ensure a unique `data_transfer_direction` or should we be grouping by the `data_transfer_direction` result as well?
```
GROUP BY aws.lineitem_usagestartdate,
    aws.lineitem_resourceid,
    4, -- CASE satement
    aws.product_productfamily,
    aws.product_instancetype,
    aws.lineitem_availabilityzone,
    aws.product_region,
    aws.resourcetags,
    aws.costcategory
```

Reply:
We do probably need to group by the direction, but we can't use an aggregate in a group by. Grouping by `aws.lineitem_usagetype` may work.

Reply:
You can also use numbers to group by items in the `SELECT` statement, for example the `4` in the group by there is based on the `CASE` statement in the SQL